### DISCUSSION TITLE
How to use RaisePropertyChanged for DirectProperty in Avalonia 11?

### DISCUSSION THREAD

**Discussion by SwiftExtender** [AvaloniaUI/Avalonia#17286] [General]
As i see, RaisePropertyChanged first argument must be DirectPropertyBase, not DirectProperty. I cant find any example of usage RaisePropertyChanged in Avalonia 11. Please help

**Comment by maxkatz6** [comment]
Not sure if I understand the question. DirectProperty inherits DirectPropertyBase, so you should be able to use it with current RaisePropertyChanged  method.

**Comment by SwiftExtender** [comment]
Error CS0411 in code, i cant understand what i`am doing wrong

private void OnTagItemPropertyChanged(object sender, AvaloniaPropertyChangedEventArgs e)
{
    RaisePropertyChanged(IsAnyItemInEditModeProperty, new Data.Optional<bool>(), ItemsSource.Any(x => x.IsInEditMode));
}

![изображение](https://github.com/user-attachments/assets/236ab052-94bd-4354-a932-6a33b5415d28)

**Reply by maxkatz6** [reply]
You are passing wrong parameters
https://github.com/AvaloniaUI/Avalonia/blob/c66008b627926b0cd35abcc83d1bf681e416b5a8/src/Avalonia.Base/AvaloniaObject.cs#L744-L747

**Comment by SwiftExtender** [comment]
Seems like this is correct
```
private void OnTagItemPropertyChanged(object sender, AvaloniaPropertyChangedEventArgs e)
{
    RaisePropertyChanged<bool>(IsAnyItemInEditModeProperty, new bool(), ItemsSource.Any(x => x.IsInEditMode));
}
```
